waku/v2/protocol/store/message_queue.go

```diff
@@ -1,11 +1,39 @@
 package store
 
+import (
+	"sync"
+	"time"
+
+	"github.com/status-im/go-waku/waku/v2/utils"
+)
+
 type MessageQueue struct {
+	sync.RWMutex
+
+	seen        map[[32]byte]struct{}
```

Comment:
could messages be a map directly? we wouldn't have to maintain 2 entry?

Reply:
The somewhat difficult part of using a single map instead of a map+slice would be to efficiently iterate the messages for cleaning up older records, as well as keeping the message queue limited to N records, because maps do not preserve order.

Reply:
I missed that part, good call. Indeed the need is an ordered Hash map, in order to build an ordered set (set by default doesn't preserve order). it seems overkill suddenly!